backend/core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/RequestHandlerBindingAssistant.java

```diff
@@ -98,14 +107,25 @@ private void assistConnectionAndBindingIfNeeded(AssistCreatingConnectionParams c
           var assistNewConnectionResult = assistCreatingConnectionAndWaitForRepositoryUpdate(connectionParams, cancelMonitor);
           var assistNewBindingResult = assistBindingAndWaitForRepositoryUpdate(assistNewConnectionResult.getNewConnectionId(), isSonarCloud,
             projectKey, cancelMonitor);
-          callback.andThen(assistNewConnectionResult.getNewConnectionId(), assistNewBindingResult.getConfigurationScopeId(), cancelMonitor);
+          var boundScopes = new HashSet<String>();
+          if (assistNewBindingResult.getConfigurationScopeId() != null) {
+            boundScopes.add(assistNewBindingResult.getConfigurationScopeId());
+          }
+          callback.andThen(assistNewConnectionResult.getNewConnectionId(), boundScopes, assistNewBindingResult.getConfigurationScopeId(), cancelMonitor);
         } finally {
           endFullBindingProcess();
         }
       } else {
-        // we pick the first connection but this could lead to issues later if there were several matches (make the user select the right
-        // one?)
-        assistBindingIfNeeded(connectionsMatchingOrigin.get(0).getConnectionId(), isSonarCloud, projectKey, callback, cancelMonitor);
+        var isOriginTrusted = repository.hasConnectionWithOrigin(origin);
+        if (isOriginTrusted) {
```

Comment:
Shouldn't this check be moved earlier? I assume we would like to check the origin also for the if branch above

Reply:
We cannot check the origin if the connection has not yet been created because we don't know what a trusted connection is. During the connection set-up, a warning already exists to warn users about malicious servers. The issue is when the connection already exists.